addons/hr_fleet/wizard/hr_departure_wizard.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class HrDepartureWizard(models.TransientModel):
+    _inherit = 'hr.departure.wizard'
+
+    release_campany_car = fields.Boolean("Release Company Car", default=True)
+
+    def action_register_departure(self):
+        super(HrDepartureWizard, self).action_register_departure()
+        if self.release_campany_car:
+            self._free_campany_car()
+
+    def _free_campany_car(self):
+        """Find all fleet.vehichle.assignation.log records that link to the employee. For each record, if the current driver
+        is still the employee, set it to False. If there is no end date or end date > departure date, update the date."""
+        drivers = self.employee_id.user_id.partner_id | self.employee_id.sudo().address_home_id
+        assignations = self.env['fleet.vehicle.assignation.log'].search([('driver_id', 'in', drivers.ids)])
```

Comment:
I'm not sure that all the cars have a related ``fleet.vehicle.assignation.log`` as this model has been introduced later.

Maybe we could use the ``assignations`` recordset to set the date end + make an additional search on the ``fleet.vehicle`` based on the ``driver_id`` field to set the value to ``False``.

What do you think ?